src/Installer/dnup/DotnetInstallManager.cs

```diff
@@ -36,7 +36,7 @@ public DotnetInstallManager(IEnvironmentProvider? environmentProvider = null)
         string programFiles = Environment.GetFolderPath(Environment.SpecialFolder.ProgramFiles);
         string programFilesX86 = Environment.GetFolderPath(Environment.SpecialFolder.ProgramFilesX86);
         bool isAdminInstall = installDir.StartsWith(Path.Combine(programFiles, "dotnet"), StringComparison.OrdinalIgnoreCase) ||
-                              installDir.StartsWith(Path.Combine(programFilesX86, "dotnet"), StringComparison.OrdinalIgnoreCase);
+                              installDir.StartsWith(Path.Combine(programFilesX86, "dotnet"), StringComparison.OrdinalIgnoreCase); // TODO: This should be improved to not be windows-specific
```

Comment:
Make sure we have issues filed for TODOs that are important.

Reply:
https://github.com/dotnet/sdk/issues/51601 great point, filed